src/Juvix/Compiler/Nockma/Translation/FromTree.hs

```diff
@@ -469,38 +517,44 @@ compile = \case
             path :: Sem r Path
             path = do
               fr <- directRefPath _fieldRef
-              return $ case memrep of
-                NockmaMemRepConstr ->
-                  fr
-                    ++ constructorPath ConstructorArgs
-                    ++ indexStack argIx
-                NockmaMemRepTuple ->
-                  fr
-                    ++ indexTuple
-                      IndexTupleArgs
-                        { _indexTupleArgsLength = arity,
-                          _indexTupleArgsIndex = argIx
-                        }
-                NockmaMemRepList constr -> case constr of
-                  NockmaMemRepListConstrNil -> impossible
-                  NockmaMemRepListConstrCons ->
-                    fr
-                      ++ indexTuple
+              return $
+                fr
+                  ++ case memrep of
+                    NockmaMemRepConstr ->
+                      constructorPath ConstructorArgs
+                        ++ indexStack argIx
+                    NockmaMemRepTuple ->
+                      indexTuple
                         IndexTupleArgs
-                          { _indexTupleArgsLength = 2,
+                          { _indexTupleArgsLength = arity,
                             _indexTupleArgsIndex = argIx
                           }
-                NockmaMemRepMaybe constr -> case constr of
-                  NockmaMemRepMaybeConstrNothing -> impossible
-                  -- just x is represented as [nil x] so argument index is offset by 1.
-                  -- argIx will always be 0 because just has one argument
-                  NockmaMemRepMaybeConstrJust ->
-                    fr
-                      ++ indexTuple
-                        IndexTupleArgs
-                          { _indexTupleArgsLength = 2,
-                            _indexTupleArgsIndex = argIx + 1
-                          }
+                    NockmaMemRepList constr -> case constr of
+                      NockmaMemRepListConstrNil -> impossible
+                      NockmaMemRepListConstrCons ->
+                        indexTuple
+                          IndexTupleArgs
+                            { _indexTupleArgsLength = 2,
+                              _indexTupleArgsIndex = argIx
+                            }
+                    NockmaMemRepNoun n -> case n of
+                      NockmaMemRepAtom -> []
```

Comment:
Perhaps use `emptyPath` from `Juvix.Compiler.Nockma.Language.Path` instead of `[]`?

Reply:
done